script.js

```diff
@@ -2,11 +2,104 @@
 function setup() {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  selectEpisode(allEpisodes);
 }
+// converting objects into an
+// episodes = Object.entries(getAllEpisodes())
+// console.log(episodes);
+function makePageForEpisodes(allEpisodes) {
+  allEpisodes.forEach((episode, index) => {
+    // Create the HTML elements needed for a card
+    const div = document.createElement("div");
+    const h4TagElem = document.createElement("h3");
+    h4TagElem.classList = "episode-info";
+    const img = document.createElement("img");
+    const pElem = document.createElement("p");
+    // the body element so that the newly created elements
+    // can be appended to it
+    const rootElem = document.getElementById("root");
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+    // Use innerHTML to add text to the anchor tag
+    h4TagElem.innerHTML =
+      `${episode.name}` +
+      `-S${episode.season.toString().padStart(2, "0")}
+        E${episode.number.toString().padStart(2, "0")}`;
+    pElem.innerHTML = episode.summary;
+    // Append the elements to the DOM
+    rootElem.appendChild(div);
+    div.append(h4TagElem, img, pElem);
+    // use setAttribute to add the card class to the div
+
+    div.setAttribute(
+      "class",
+      " main-wrapper card html body main-content card-block img"
+    );
+    //div.setAttribute("class", "cards")
+    img.setAttribute("src", episode.image.medium);
+  });
+}
+//filter episodes
+const searchBar = document.querySelector("input");
+searchBar.addEventListener("keyup", filterSearch);
+let main = document.querySelector("#root");
+function filterSearch(event) {
+  let firstValue = event.target.value.toLowerCase();
+  /* done with Orhan's help*/
+  let episodes = [...main.children]; // making episodes into an array
+  episodes.forEach((show) => {
+    const showText = show.innerText.toLowerCase();
+    if (showText.indexOf(firstValue) < 0) {
+      show.style.display = "none";
+    } else if (showText.indexOf(firstValue.length)) {
+      show.style.display = "inline-flex";
+    }
+  });
+  let filteredSearch = episodes.filter(
+    (ep) => getComputedStyle(ep).display == "inline-flex"
```

Comment:
This works, but relying on a CSS property for data handling is a little hacky.

The approach I would take would be something like this:
- keep two global variables, `allEpisodes` and `displayedEpisodes` that indicate the state of the app and the source of truth
- when the user inputs a search query, I would look through the `allEpisodes` array, not through the episodes in the DOM
- I would then set `displayedEpisodes` according to the result of the search
- another function would take `displayedEpisodes` and update the DOM accordingly (either using your approach of setting display on and off, just re-rendering everything - not sure how fast the latter would be)